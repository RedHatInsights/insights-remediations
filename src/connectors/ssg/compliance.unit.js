'use strict';
/* eslint-disable max-len */
/* eslint-disable security/detect-non-literal-fs-filename */

const P = require('bluebird');
const fs = require('fs');
const path = require('path');
const compliance = require('./compliance');
const base = require('../../test');
const { mockRequest } = require('../testUtils');
const request = require('../../util/request');

const REQ = {
    headers: {
        'x-rh-identity': 'identity',
        'x-rh-insights-request-id': 'request-id'
    },
    identity: { type: 'test' },
    user: { username: 'test', account_number: 'test' }
};

describe('compliance impl', function () {
    beforeEach(mockRequest);

    test('returns template', async function () {
        const http = base.getSandbox().stub(request, 'run').resolves({
            statusCode: 200,
            body: [{
                template: fs.readFileSync(path.join(__dirname, 'mock', 'standard', 'service_autofs_disabled.yml'), 'utf-8'),
                version: 'unit'
            }],
            headers: {
                'x-ssg-version': 'unit'
            }
        });

        const { template, version } = await compliance.getTemplate(REQ, 'rhel7|standard|xccdf_org.ssgproject.content_rule_service_autofs_disabled');

        version.should.equal('unit');
        expect(template).toMatchSnapshot();
        http.callCount.should.equal(1);
    });

    test('returns template for multiple issues', async function () {
        const http = base.getSandbox().stub(request, 'run').resolves({
            statusCode: 200,
            body: [{
                template: fs.readFileSync(path.join(__dirname, 'mock', 'standard', 'service_autofs_disabled.yml'), 'utf-8'),
                version: 'unit'
            }, {
                template: fs.readFileSync(path.join(__dirname, 'mock', 'standard', 'service_rsyslog_enabled.yml'), 'utf-8'),
                version: 'unit'
            }],
            headers: {
                'x-ssg-version': 'unit'
            }
        });

        const pending = [
            compliance.getTemplate(REQ, 'rhel7|standard|xccdf_org.ssgproject.content_rule_service_autofs_disabled'),
            compliance.getTemplate(REQ, 'rhel7|standard|xccdf_org.ssgproject.content_rule_service_rsylog_enabled.yml')
        ];

        const results = await P.all(pending);

        results[0].version.should.equal('unit');
        results[1].version.should.equal('unit');
        expect(results[0].template).toMatchSnapshot();
        expect(results[1].template).toMatchSnapshot();
        http.callCount.should.equal(1);
    });
});
