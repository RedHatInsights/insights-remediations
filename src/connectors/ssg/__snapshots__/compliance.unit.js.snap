// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compliance impl returns template 1`] = `
"# platform = multi_platform_all
# complexity = low
# strategy = disable
# reboot = false
# disruption = low
- name: Disable the Automounter
  hosts: '@@HOSTS@@'
  become: true
  tags:
    - CCE-27498-5
    - DISA-STIG-RHEL-07-020110
    - NIST-800-171-3.4.6
    - NIST-800-53-AC-19(a)
    - NIST-800-53-AC-19(d)
    - NIST-800-53-AC-19(e)
    - NIST-800-53-IA-3
    - disable_strategy
    - low_complexity
    - low_disruption
    - medium_severity
    - no_reboot_needed
    - service_autofs_disabled
  tasks:

    - name: Unit Service Exists - autofs.service
      command: systemctl list-unit-files autofs.service
      register: service_file_exists
      changed_when: false
      ignore_errors: true
      when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type
        != \\"docker\\"

    - name: Disable service autofs
      systemd:
        name: autofs.service
        enabled: 'no'
        state: stopped
        masked: 'yes'
      when:
        - '\\"autofs.service\\" in service_file_exists.stdout_lines[1]'
        - ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type !=
          \\"docker\\"

    - name: Unit Socket Exists - autofs.socket
      command: systemctl list-unit-files autofs.socket
      register: socket_file_exists
      changed_when: false
      ignore_errors: true
      when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type
        != \\"docker\\"

    - name: Disable socket autofs
      systemd:
        name: autofs.socket
        enabled: 'no'
        state: stopped
        masked: 'yes'
      when:
        - '\\"autofs.socket\\" in socket_file_exists.stdout_lines[1]'
        - ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type !=
          \\"docker\\"
"
`;

exports[`compliance impl returns template for multiple issues 1`] = `
"# platform = multi_platform_all
# complexity = low
# strategy = disable
# reboot = false
# disruption = low
- name: Disable the Automounter
  hosts: '@@HOSTS@@'
  become: true
  tags:
    - CCE-27498-5
    - DISA-STIG-RHEL-07-020110
    - NIST-800-171-3.4.6
    - NIST-800-53-AC-19(a)
    - NIST-800-53-AC-19(d)
    - NIST-800-53-AC-19(e)
    - NIST-800-53-IA-3
    - disable_strategy
    - low_complexity
    - low_disruption
    - medium_severity
    - no_reboot_needed
    - service_autofs_disabled
  tasks:

    - name: Unit Service Exists - autofs.service
      command: systemctl list-unit-files autofs.service
      register: service_file_exists
      changed_when: false
      ignore_errors: true
      when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type
        != \\"docker\\"

    - name: Disable service autofs
      systemd:
        name: autofs.service
        enabled: 'no'
        state: stopped
        masked: 'yes'
      when:
        - '\\"autofs.service\\" in service_file_exists.stdout_lines[1]'
        - ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type !=
          \\"docker\\"

    - name: Unit Socket Exists - autofs.socket
      command: systemctl list-unit-files autofs.socket
      register: socket_file_exists
      changed_when: false
      ignore_errors: true
      when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type
        != \\"docker\\"

    - name: Disable socket autofs
      systemd:
        name: autofs.socket
        enabled: 'no'
        state: stopped
        masked: 'yes'
      when:
        - '\\"autofs.socket\\" in socket_file_exists.stdout_lines[1]'
        - ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type !=
          \\"docker\\"
"
`;

exports[`compliance impl returns template for multiple issues 2`] = `
"# platform = multi_platform_all
# complexity = low
# strategy = enable
# reboot = false
# disruption = low
- name: Enable rsyslog Service
  hosts: '@@HOSTS@@'
  become: true
  tags:
    - CCE-80188-6
    - NIST-800-53-AU-12
    - NIST-800-53-AU-4(1)
    - enable_strategy
    - low_complexity
    - low_disruption
    - medium_severity
    - no_reboot_needed
    - service_rsyslog_enabled
  tasks:

    - name: Enable service rsyslog
      block:

        - name: Gather the package facts
          package_facts:
            manager: auto

        - name: Enable service rsyslog
          service:
            name: rsyslog
            enabled: 'yes'
            state: started
          when:
            - '\\"rsyslog\\" in ansible_facts.packages'
      when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type
        != \\"docker\\"
"
`;
